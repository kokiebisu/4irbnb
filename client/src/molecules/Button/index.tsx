import React from 'react';

// atoms
import { Button } from 'atoms';

// blocks
import SearchButton from 'molecules/Button/button.search';
import CookieButton from 'molecules/Button/button.cookie';
import GlobeButton from 'molecules/Button/button.globe';

export type ButtonProps = {
  type?: string;
  onPress: () => void;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  inverse?: boolean;
  name?: string;
};

interface mapProps {
  [key: string]: JSX.Element;
}

/**
 * Use a button when you want to perform specific actions
 **/
export const CustomButton = ({ type, ...props }: ButtonProps) => {
  if (!type) {
    return <Button {...props} />;
  }
  const types: mapProps = {
    search: <SearchButton {...props} />,
    cookie: <CookieButton {...props} />,
    globe: <GlobeButton {...props} />,
  };
  return types[type];
};
