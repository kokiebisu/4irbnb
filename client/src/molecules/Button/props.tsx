export type Props = {
  type?: string;
  onPress: () => void;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  inverse?: boolean;
  name?: string;
};
