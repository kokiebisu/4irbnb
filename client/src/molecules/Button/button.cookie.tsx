import React from 'react';
import { css } from 'styled-components';
import theme from 'styled-theming';

// atoms
import { Button, Text } from 'atoms';

// colors
import { colors } from 'styles';

// type
import { ButtonProps } from 'molecules/Button';

/**
 - Use a button when you want to perform specific cookie actions
 **/
export default ({ inverse, name, ...props }: ButtonProps) => {
  return (
    <Button
      styles={inverse ? extend.inverse.button : extend.normal.button}
      {...props}>
      <Text styles={inverse ? extend.inverse.text : extend.normal.text}>
        {name}
      </Text>
    </Button>
  );
};

const primary = theme('mode', {
  light: colors.bluegreen,
});

const secondary = theme('mode', {
  light: colors.white,
});

const common = css`
  padding: 15px 0;
  font-size: 12px;
  font-weight: 700;
  min-width: 125px;
  width: 100%;
`;

const extend = {
  normal: {
    button: css`
      ${common}
      background-color: ${primary};
    `,
    text: css`
      color: ${secondary};
    `,
  },
  inverse: {
    button: css`
      ${common}
      background-color: ${secondary}; 
      border: 1px solid ${primary};
    `,
    text: css`
      color: ${primary};
    `,
  },
};
