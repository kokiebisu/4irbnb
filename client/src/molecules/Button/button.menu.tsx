import React from 'react';
import styled, { css } from 'styled-components';
import { Box, Icon, Button } from 'atoms';
import { colors } from 'styles';
import { CustomProps } from 'molecules/Button';
import theme from 'styled-theming';
import { lighten } from 'polished';

const menuButtonColors = {
  background: theme('mode', {
    light: colors.white,
  }),
  content: theme('mode', {
    light: colors.gray,
  }),
  border: theme('mode', {
    light: lighten(0.4, colors.gray),
  }),
};

const menuButtonStyles = {
  wrapper: css`
    background-color: ${menuButtonColors.background};
    border-radius: 48px;
    padding: 5px 6px 5px 12px;
    transition: 0.2s ease-in;
    border: 1px solid ${menuButtonColors.border};
    &:hover {
      box-shadow: rgba(0, 0, 0, 0.15) 0px 2px 8px;
    }
  `,
  flex: css`
    display: flex;
    align-items: center;
  `,
  icon: css`
    svg {
      width: 100%;
      height: 100%;
      path {
        fill: ${menuButtonColors.content};
      }
    }
  `,
  bars: css`
    margin-right: 12px;
    height: 15px;
    width: 17px;
  `,
  avatar: css`
    height: 32px;
    width: 32px;
  `,
};

export const MenuButton: React.FC<CustomProps> = ({ name, ...props }) => {
  return (
    <Button styles={menuButtonStyles.wrapper} {...props}>
      <Box styles={menuButtonStyles.flex}>
        <Icon
          styles={[...menuButtonStyles.bars, ...menuButtonStyles.icon]}
          name='Bars'
        />
        <Icon
          styles={[...menuButtonStyles.avatar, ...menuButtonStyles.icon]}
          name='Avatar'
        />
      </Box>
    </Button>
  );
};
