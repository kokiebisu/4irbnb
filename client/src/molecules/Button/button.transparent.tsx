import React from 'react';
import { css } from 'styled-components';
import { Button } from 'atoms';
import { colors } from 'styles';
import { CustomProps } from 'molecules/Button';
import theme from 'styled-theming';

export const transparentButtonColors = {
  stroke: theme('mode', {
    light: colors.white,
  }),
  hover: theme('mode', {
    light: 'rgba(190, 190, 190, 0.2)',
  }),
};

const transparentButtonStyles = {
  wrapper: css`
    padding: 15px;
    background-color: transparent;
    border-radius: 24px;
    cursor: pointer;
    &:focus {
      outline: none;
    }
    &:hover {
      background-color: ${transparentButtonColors.hover};
    }
  `,
};

export const globeButtonStyles = {
  flex: css`
    display: flex;
    align-items: center;
  `,
  globe: css`
    position: relative;
    top: 50%;
    margin-right: 6px;
    width: 16px;
    height: 16px;
  `,
  arrow: css`
    width: 10px;
  `,
};

export const hostButtonStyles = {};

export const TransparentButton: React.FC<CustomProps> = ({
  name,
  children,
  ...props
}) => {
  return (
    <Button styles={transparentButtonStyles.wrapper} {...props}>
      {children}
    </Button>
  );
};
