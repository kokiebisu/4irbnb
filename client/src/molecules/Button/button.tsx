import React from 'react';

// atoms
import { BaseButton } from 'atoms/Button';

// blocks
import SearchButton from 'molecules/Button/button.search';
import CookieButton from 'molecules/Button/button.cookie';
import GlobeButton from 'molecules/Button/button.globe';

export type ButtonProps = {
  type?: string;
  onPress: () => void;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  inverse?: boolean;
  name?: string;
};

interface mapProps {
  [key: string]: JSX.Element;
}

/**
 * Use a button when you want to perform specific actions
 **/
export const Button = ({ type, ...props }: ButtonProps) => {
  if (!type) {
    return <BaseButton {...props} />;
  }
  const types: mapProps = {
    search: <SearchButton {...props} />,
    cookie: <CookieButton {...props} />,
    globe: <GlobeButton {...props} />,
  };
  return types[type];
};
