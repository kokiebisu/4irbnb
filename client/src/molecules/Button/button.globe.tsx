import React from 'react';
import { css } from 'styled-components';

// atoms
import { Icon } from 'atoms/Icon';
import { BaseButton } from 'atoms/Button';
import { Box } from 'atoms/Box';

import { Flex } from 'layout/flex';

// type
import { ButtonProps } from 'molecules/Button/button';
import theme from 'styled-theming';

import { colorpallete } from 'styles/colorpallete';

export default ({ name, ...props }: ButtonProps) => {
  return (
    <BaseButton styles={extend.wrapper} {...props}>
      <Flex alignCenter>
        <Icon styles={extend.icon.globe} name='Globe' />
        <Icon styles={extend.icon.arrow} name='BottomArrow' />
      </Flex>
    </BaseButton>
  );
};

const stroke = theme('mode', {
  light: colorpallete.white,
});

const extend = {
  wrapper: css`
    padding: 10px 14px;
    background-color: transparent;
    border-radius: 24px;
    cursor: pointer;
    &:focus {
      outline: none;
    }
    &:hover {
      background-color: rgba(190, 190, 190, 0.2);
    }
  `,
  icon: {
    globe: css`
      position: relative;
      top: 50%;
      margin-right: 6px;
      width: 16px;
      height: 16px;
      svg {
        width: 100%;
        path {
          fill: ${stroke};
        }
      }
    `,
    arrow: css`
      width: 10px;
      svg {
        width: 100%;
        polygon {
          fill: ${stroke};
        }
      }
    `,
  },
};
