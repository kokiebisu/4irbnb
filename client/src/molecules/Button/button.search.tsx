import React from 'react';
import { css } from 'styled-components';

// breakpoints
import { sizes } from '../../styles/breakpoints';

// atoms
import { Icon } from 'atoms/Icon';
import { BaseButton } from 'atoms/Button';
import { Text } from 'atoms/Text';
import { Box } from 'atoms/Box';

// type
import { ButtonProps } from 'molecules/Button/button';

export default ({ name, ...props }: ButtonProps) => {
  return (
    <BaseButton styles={extend.button} {...props}>
      <Box styles={extend.box}>
        <Icon styles={extend.icon} name='search' />
        <Text styles={extend.text}>{name}</Text>
      </Box>
    </BaseButton>
  );
};

const extend = {
  button: css`
    background: -webkit-linear-gradient(
      right,
      rgb(230, 30, 77) 0%,
      rgb(227, 28, 95) 50%,
      rgb(215, 4, 102) 100%
    );
    border: none;
    color: white;
  `,
  box: css`
    display: block;
    @media ${sizes.sm} {
      display: flex;
      align-items: center;
    }
  `,
  icon: css`
    position: relative;
    top: 1.5px;

    & svg {
      width: 14px;
      height: 14px;
      stroke: white;
      path {
        fill: white;
        stroke-width: 2;
      }
    }
  `,
  text: css`
    display: none;
    font-size: 15px;
    letter-spacing: 0.5px;
    font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto,
      Helvetica Neue, sans-serif !important;
    font-weight: 400;
    padding-left: 7px;

    @media ${sizes.sm} {
      display: block;
    }
  `,
};
