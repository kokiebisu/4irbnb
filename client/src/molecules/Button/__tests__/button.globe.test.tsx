import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// atoms
import { CustomButton } from 'molecules/Button';

describe('Cookie Button', () => {
  it('renders correctly', () => {
    const testingData = {
      action: () => console.log('test'),
    };

    const tree = renderer
      .create(
        <CustomButton
          type='globe'
          onPress={testingData.action}
          {...testingData}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
