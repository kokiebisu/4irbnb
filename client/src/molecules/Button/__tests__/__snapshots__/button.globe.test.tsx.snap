// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Globe Button renders correctly 1`] = `
.c0 {
  border-radius: 8px;
  font-size: 18px;
  border: none;
  padding: 10px 14px;
  background-color: transparent;
  border-radius: 24px;
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  background-color: rgba(190,190,190,0.2);
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  top: 50%;
  margin-right: 6px;
  width: 16px;
  height: 16px;
}

.c2 svg {
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 10px;
}

.c3 svg {
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  action={[Function]}
  className="c0"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      style={Object {}}
    >
      <svg
        viewBox="0 0 16 16"
      >
        <path
          d="m8.002.25a7.77 7.77 0 0 1 7.748 7.776 7.75 7.75 0 0 1 -7.521 7.72l-.246.004a7.75 7.75 0 0 1 -7.73-7.513l-.003-.245a7.75 7.75 0 0 1 7.752-7.742zm1.949 8.5h-3.903c.155 2.897 1.176 5.343 1.886 5.493l.068.007c.68-.002 1.72-2.365 1.932-5.23zm4.255 0h-2.752c-.091 1.96-.53 3.783-1.188 5.076a6.257 6.257 0 0 0 3.905-4.829zm-9.661 0h-2.75a6.257 6.257 0 0 0 3.934 5.075c-.615-1.208-1.036-2.875-1.162-4.686l-.022-.39zm1.188-6.576-.115.046a6.257 6.257 0 0 0 -3.823 5.03h2.75c.085-1.83.471-3.54 1.059-4.81zm2.262-.424c-.702.002-1.784 2.512-1.947 5.5h3.904c-.156-2.903-1.178-5.343-1.892-5.494l-.065-.007zm2.28.432.023.05c.643 1.288 1.069 3.084 1.157 5.018h2.748a6.275 6.275 0 0 0 -3.929-5.068z"
        />
      </svg>
    </div>
    <div
      className="c3"
      style={Object {}}
    >
      <svg
        viewBox="0 0 256 256"
      >
        <g>
          <g>
            <polygon
              points="225.813,48.907 128,146.72 30.187,48.907 0,79.093 128,207.093 256,79.093 		"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>
</button>
`;
